src/main/java/com/redhat/devtools/lsp4ij/features/foldingRange/LSPFoldingRangeBuilder.java

```diff
@@ -54,8 +66,30 @@ protected void buildLanguageFoldRegions(@NotNull List<FoldingDescriptor> descrip
         }
 
         PsiFile file = root.getContainingFile();
+        List<FoldingRange> foldingRanges = getFoldingRanges(file);

```

Comment:
Again, the LSP client logic here has been encapsulated into this reusable method.